### ISSUE TITLE
Non-matching glob causes segfault

### ISSUE THREAD

**Issue by aidanheerdegen** [coecms/mppnccombine-fast#36]
Run program with glob that matches no files and no error is returned. Should match what happens if no files specified on command line:

```
$ mpirun -n 2 /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast -o ocean_month.nc 'blahblahblah*'
mpirun: INFO: switching to OpenMPI runtime version 3.0.1
Copying non-collated variables
[raijin3:27617] *** Process received signal ***
[raijin3:27617] Signal: Segmentation fault (11)
[raijin3:27617] Signal code: Address not mapped (1)
[raijin3:27617] Failing at address: (nil)
[raijin3:27617] [ 0] /lib64/libpthread.so.0(+0xf7e0)[0x2b2fca8437e0]
[raijin3:27617] [ 1] /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast[0x403d03]
[raijin3:27617] [ 2] /lib64/libc.so.6(__libc_start_main+0x100)[0x2b2fcaa6fd20]
[raijin3:27617] [ 3] /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast[0x403889]
[raijin3:27617] *** End of error message ***
-------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
-------------------------------------------------------
--------------------------------------------------------------------------
mpirun noticed that process rank 0 with PID 0 on node raijin3 exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
```

Get the same behaviour with an unquote `glob`, which I guessed gets passed as an input filename

```
$ mpirun -n 2 /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast -o ocean_month.nc blahblahblah* 
mpirun: INFO: switching to OpenMPI runtime version 3.0.1
Copying non-collated variables
[raijin3:10663] *** Process received signal ***
[raijin3:10663] Signal: Segmentation fault (11)
[raijin3:10663] Signal code: Address not mapped (1)
[raijin3:10663] Failing at address: (nil)
[raijin3:10663] [ 0] /lib64/libpthread.so.0(+0xf7e0)[0x2b156baa17e0]
[raijin3:10663] [ 1] /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast[0x403d03]
[raijin3:10663] [ 2] /lib64/libc.so.6(__libc_start_main+0x100)[0x2b156bccdd20]
[raijin3:10663] [ 3] /home/502/aph502/code/c/mppnccombine-fast/mppnccombine-fast[0x403889]
[raijin3:10663] *** End of error message ***
-------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
-------------------------------------------------------
--------------------------------------------------------------------------
mpirun noticed that process rank 0 with PID 0 on node raijin3 exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
```

**Comment by ScottWales** [comment]
Fixed in beta branch #33

**Comment by aidanheerdegen** [comment]
Confirmed fixed.
